benchmark/benchresult/main.go

```diff
@@ -0,0 +1,135 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"encoding/gob"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/benchmark/stats"
+)
+
+func createMap(fileName string, m map[string]stats.BenchResults) {
+	f, err := os.Open(fileName)
+	if err != nil {
+		log.Fatalf("Read file %s error: %s\n", fileName, err)
+	}
+	defer f.Close()
+	var data []stats.BenchResults
+	decoder := gob.NewDecoder(f)
+	if err = decoder.Decode(&data); err != nil {
+		log.Fatalf("Decode file %s error: %s\n", fileName, err)
+	}
+	for _, d := range data {
+		m[d.RunMode+"-"+d.Features.String()] = d
+	}
+}
+
+func getPercentage(val1, val2, factor float64) string {
+	return strconv.FormatFloat((val2-val1*factor)*100.0/(val1*factor), 'f', 2, 64) + "% "
+}
+
+func intChange(title string, val1, val2 int64, factor float64) string {
+	percentChange := getPercentage(float64(val1), float64(val2), factor)
+	return fmt.Sprintf("%10s  %12s  %12s   %8s \n", title, strconv.FormatInt(val1, 10),
+		strconv.FormatInt(val2, 10), percentChange)
+}
+
+func timeChange(title int, val1, val2 time.Duration, factor float64) string {
+	percentChange := getPercentage(float64(val1), float64(val2), factor)
```

Comment:
Why isn't ((val2-val1)*100)/(val1) sufficient?  If these are true time.Durations (int64 nanos), this should "just work" without anything special.

Reply:
Done. My old version is read those data as string, factor is used when data has different unit.